src/Features/Core/Portable/NavigateTo/NavigateToSearcher.cs

```diff
@@ -94,16 +126,19 @@ internal async Task SearchAsync(bool searchCurrentDocument, CancellationToken ca
                     // totally hydrated the oop side.  Until that happens, we'll attempt to return cached data from languages
                     // that support that.
                     isFullyLoaded = await _host.IsFullyLoadedAsync(cancellationToken).ConfigureAwait(false);
-                    await SearchAllProjectsAsync(isFullyLoaded, cancellationToken).ConfigureAwait(false);
+                    await SearchAllProjectsAsync(isFullyLoaded, scope, cancellationToken).ConfigureAwait(false);
                 }
             }
             catch (OperationCanceledException)
             {
             }
             finally
             {
-                // providing this extra information will make UI to show indication to users
-                // that result might not contain full data
+                // Ensure that we actually complete all our remaining progress items so that the progress bar completes.
+                await ProgressItemsCompletedAsync(_remainingProgressItems, cancellationToken).ConfigureAwait(false);
+                Debug.Assert(_remainingProgressItems == 0);
```

Comment:
it was becoming a PITA for all the control flow to keep track of this and set progress to the 'finished' state at teh end.  So now we just keep track ourselves across all codepaths and just jump to teh end unilaterally.